ITensorGaussianMPS/src/gmps.jl

```diff
@@ -333,6 +732,12 @@ function slater_determinant_to_mps(s::Vector{<:Index}, Φ::AbstractMatrix; kwarg
   return correlation_matrix_to_mps(s, conj(Φ) * transpose(Φ); kwargs...)
 end
 
+function slater_determinant_to_mps(
+  s::Vector{<:Index}, Φ::T; kwargs...
+) where {T<:Union{Pairing,ConservingNf}}
+  return correlation_matrix_to_mps(s, T(conj(Φ.data) * transpose(Φ.data)); kwargs...)
```

Comment:
I think for things like `conj` and `transpose` we should define overloads for `ConservesNf`/`ConservesNfParity`. That would make this code a bit nicer.

Reply:
I agree it makes sense to have these overloads, but in this specific case it doesn't help much unless we want to implement most of the AbstractMatrix Interface (* requires unwrapping)

Reply:
Ah right, I guess once you get `*` involved it gets more complicated. You can still define something relatively generic like:
```julia
(x::AbstractSymmetry * y::AbstractSymmetry) = set_data(x, x.data * y.data)
```

Reply:
Could be done. For this concrete example, I don't particularly like that that one doesn't error if x,y are different concrete types of AbstractSymmetry.

Reply:
Yeah, I was thinking that too. You could restrict them to have exactly the same type like this:
```julia
(x::T * y::T) where {T<:AbstractSymmetry} = set_data(x, x.data * y.data)
```
but that is already too strict since even in this case one is transposed.

We could do:
```julia
function (x::AbstractSymmetry * y::AbstractSymmetry)
  if !has_same_symmetry(x, y)
    error("Can't multiply two symmetric objects with different symmetries.")
  end
  return set_data(x, x.data * y.data)
end

has_same_symmetry(::AbstractSymmetry, ::AbstractSymmetry) = false
has_same_symmetry(::ConservesNf, ::ConservesNf) = true
has_same_symmetry(::ConservesNfParity, ::ConservesNfParity) = true
```
That's not such a burden to require of new symmetry types.